### ISSUE TITLE
how to parse the  heatmap binaries Openpose saved？

### ISSUE THREAD

**Issue by lunalulu** [CMU-Perceptual-Computing-Lab/openpose#1333]
the file like this :
000000000009_pose_heatmaps.float

thanks very much!

**Comment by lunalulu** [comment]
My command:
./build/examples/openpose/openpose.bin --image_dir xxx --heatmaps_add_parts --heatmaps_add_PAFs --write_heatmaps xxx --net_resolution 368x368 --heatmaps_scale 0 --write_heatmaps_format float

**Comment by lunalulu** [comment]
@gineshidalgo99 thanks for your reply!
i have paras the .float file with the code below:
But there's a displacement when I fuse the part heatmaps and paf into the original picture.
i don't know why?
code:
`
import cv2
import numpy as np
import struct
import os

f = open('000000000382_pose_heatmaps.float', 'rb')#save parts heatmaps and paf with size=(368*368)
nx = 28336
ny = 368
pic = np.zeros((nx, ny))#float type
pic_unit8 = np.zeros((nx, ny), dtype=np.uint8)
f.seek(16)
for i in range(nx):
    for j in range(ny):
        data = f.read(4)
        elem = struct.unpack('f', data)[0]
        #print(struct.unpack('f', data))
        pic[i][j] = elem
f.close()

#convert to unit8
for h in range(pic.shape[0]):
    for w in range(pic.shape[1]):
        if pic[h,w] <= 0:
            pic_unit8[h,w] = 0
        else:
            pic_unit8[h,w] = int(pic[h,w]*255)

OUT_PATH = 'mixImages'

image = cv2.imread('000000000382.jpg')
cv2.imshow('image_ori', image)
image = cv2.resize(image,  (368, 368), interpolation=cv2.INTER_CUBIC)

#heatmap0
img = pic_unit8[0:368, 0:368]

#convert to three channels
img_heatmap0 = np.expand_dims(img, axis=2)
img_heatmap0 = np.concatenate((img_heatmap0, img_heatmap0, img_heatmap0), axis=-1)

image_heatmap_mix = np.zeros((image.shape[1], image.shape[0]), dtype=np.uint8)

for i in range(1,25): #show all parts heatmaps
    print(i)
    a = 368*i
    b = 368*(i+1)

    img = pic_unit8[a:b, 0:368]

    #convert to three channels
    img_heatmap = np.expand_dims(img, axis=2)
    img_heatmap = np.concatenate((img_heatmap, img_heatmap, img_heatmap), axis=-1)

    if i == 1:
        image_heatmap_mix = np.maximum(img_heatmap0, img_heatmap)
    else:
        image_heatmap_mix = np.maximum(image_heatmap_mix, img_heatmap)
    # save_path = os.path.join(OUT_PATH, 'heatmap'+str(i)+'.jpg')
    # print(save_path)
    # cv2.imwrite(save_path, mix)
    #cv2.imshow('mix', image)

    #cv2.waitKey(2)

save_path = os.path.join(OUT_PATH, '382_heatmap_368x368'+'.jpg')
print(save_path)
cv2.imwrite(save_path, image_heatmap_mix)

image_mix = cv2.addWeighted(image, 0.5, image_heatmap_mix, 0.5, 0.0)
save_path = os.path.join(OUT_PATH, '000000000382_heatmap_368x368'+'.jpg')
print(save_path)
cv2.imwrite(save_path, image_mix)
`
result as below:
![382_heatmap_368x368](https://user-images.githubusercontent.com/40664601/63751278-9c7ac880-c8e1-11e9-96b2-b8ca6fa3aab0.jpg)
![000000000382_heatmap_368x368](https://user-images.githubusercontent.com/40664601/63751282-9f75b900-c8e1-11e9-811e-ec695269bb7a.jpg)

thanks very much!

**Comment by lunalulu** [comment]
Maybe it's because the padding on the input image?

**Comment by lunalulu** [comment]
now, the problem has been solved. i can get the right heatmaps!
but I don't understand the way of padding.
`
padding = 0
dim_min = min(image_rigin_h, image_rigin_w)
dim_max = max(image_rigin_h, image_rigin_w)
x1 = 0
x2 = dim_max
y1 = padding
y2 = padding + dim_min
`

**Comment by gineshidalgo99** [comment]
We pad while keeping the aspect ratio

**Comment by lunalulu** [comment]
@gineshidalgo99
 i know that
but Normally we pad around the image, not only bottom and right

thank you always

**Comment by gineshidalgo99** [comment]
Oh yeah, both are possible variants, we use the former.

**Comment by lunalulu** [comment]
I suggest it be documented, if not!
thanks very much!

**Comment by gineshidalgo99** [comment]
Yes! I added it to my local doc/output.md, which will be pushed in the next commit. Thanks again.
